risc0/build/src/docker.rs

```diff
@@ -36,23 +48,30 @@ const DOCKER_IGNORE: &str = r#"
 
 const TARGET_DIR: &str = "target/riscv-guest/riscv32im-risc0-zkvm-elf/docker";
 
+/// Indicates weather the build was successful or skipped.
+pub enum BuildStatus {
+    /// The build was successful.
+    Success,
+    /// The build was skipped.
+    Skipped,
+}
+
 /// Build the package in the manifest path using a docker environment.
-pub fn docker_build(manifest_path: &Path, src_dir: &Path, features: &[String]) -> Result<()> {
+pub fn docker_build(
+    manifest_path: &Path,
+    src_dir: &Path,
+    features: &[String],
+) -> Result<BuildStatus> {
+    ensure_docker_is_running()?;
+
     if !get_env_var("RISC0_SKIP_BUILD").is_empty() {
-        return Ok(());
+        eprintln!("Skipping build because RISC0_SKIP_BUILD is set");
+        return Ok(BuildStatus::Skipped);
     }
 
-    let manifest_path = manifest_path
-        .canonicalize()
-        .context(format!("manifest_path: {manifest_path:?}"))?;
-    let src_dir = src_dir.canonicalize().context("src_dir")?;
-    eprintln!("Docker context: {src_dir:?}");
-
-    let meta = MetadataCommand::new()
-        .manifest_path(&manifest_path)
-        .exec()
-        .context("Manifest not found")?;
-    let root_pkg = meta.root_package().context("failed to parse Cargo.toml")?;
+    let manifest_path = canonicalize_path(manifest_path)?;
+    let src_dir = canonicalize_path(src_dir)?;
+    let root_pkg = get_root_pkg(&manifest_path, &src_dir)?;
```

Comment:
Why was this changed?

Reply:
I thought it would be interesting to create a helper function that could be used both in `docker.rs` and in other places if we need to get the root package. 

This was the thinking behind adding some new public functions in the file - I decided to remove them and just use the same logic in `deploy.rs` now, so the only new export that `docker.rs` has is `TARGET_DIR` and `BuildStatus`.